src/routes/+layout.server.ts

```diff
@@ -76,8 +76,11 @@ export const load: LayoutServerLoad = async ({ locals, depends }) => {
 		.limit(300)
 		.toArray();
 
+	const userAssistants = settings?.assistants?.map((assistantId) => assistantId.toString()) ?? [];
+	const userAssistantsSet = new Set(userAssistants);
```

Comment:
We have `EqualitySet` that we could bring to this codebase:

```ts

/**
 * Subclass of Set but works with non-primitives
 * using a bijective object to id mapping.
 * (passed to the constructor)
 *
 * By default the mapping is calling the String function on the
 * object.
 */
export class EqualitySet<K = ObjectId> extends Set<unknown> {
	private oToId: (o: K) => string;

	constructor(oToId?: (o: K) => string);
	constructor(iterable: Iterable<K>, oToId?: (o: K) => string);
	constructor(iterableOrFn?: Iterable<K> | ((o: K) => string), oToId?: (o: K) => string) {
		super();

		const iterable = oToId
			? (iterableOrFn as Iterable<K>)
			: iterableOrFn
				? Symbol.iterator in iterableOrFn
					? (iterableOrFn as Iterable<K>)
					: undefined
				: undefined;

		this.oToId = (iterable ? oToId : (iterableOrFn as (o: K) => string)) || String;

		/// do not pass iterable to super()
		/// because this.oToId is not yet defined.
		if (iterable) {
			for (const k of iterable) {
				this.add(k);
			}
		}
	}

	override delete(key: K): boolean {
		return super.delete(this.oToId(key));
	}

	override has(key: K): boolean {
		return super.has(this.oToId(key));
	}

	override add(key: K): this {
		return super.add(this.oToId(key));
	}

	*toIterable<Ret = K>(idToObject: (str: string) => Ret): Iterable<Ret> {
		for (const val of this) {
			yield idToObject(val as string);
		}
	}
}

```

Reply:
the process of chat-ui becoming moon-landing continues :)